src/main/java/org/sonar/plugins/csharp/CSharpSensor.java

```diff
@@ -191,6 +206,51 @@ private void importResults(Project project, SensorContext context) {
     new AnalysisResultImporter(project, context, fs, fileLinesContextFactory, noSonarFilter, perspectives).parse(analysisOutput);
   }
 
+  private void importRoslynReport(String reportPath) {
+    String contents;
+    try {
+      contents = Files.toString(new File(reportPath), Charsets.UTF_8);
+    } catch (IOException e) {
+      throw Throwables.propagate(e);
+    }
+
+    Set<String> activeRuleIds = Sets.newHashSet();
+    for (ActiveRule activeRule : ruleProfile.getActiveRulesByRepository(CSharpPlugin.REPOSITORY_KEY)) {
+      activeRuleIds.add(activeRule.getRuleKey());
+    }
+
+    JsonParser parser = new JsonParser();
+    for (JsonElement issueElement : parser.parse(contents).getAsJsonObject().get("issues").getAsJsonArray()) {
+      JsonObject issue = issueElement.getAsJsonObject();
+
+      String ruleId = issue.get("ruleId").getAsString();
+      if (activeRuleIds.contains(ruleId)) {
+        String fullMessage = issue.get("fullMessage").getAsString();
+        for (JsonElement locationElement : issue.get("locations").getAsJsonArray()) {
+          JsonObject location = locationElement.getAsJsonObject();
+          if (location.has("analysisTarget")) {
+            for (JsonElement analysisTargetElement : location.get("analysisTarget").getAsJsonArray()) {
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Refactor this code to not nest more than 4 if/for/while/switch/try statements. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AS134)